### ISSUE TITLE
Can I make the CloudFormationExecutionRole have least privilege?

### ISSUE THREAD

**Issue by pjxiao** [aws/aws-sam-cli#3282]
### Describe your idea/feature/enhancement

I hope that the CloudFormationExecutionRole have least  privilege.
Currently SAM-CLI attaches the following policy to the execution role:

```json
{
    "Version": "2012-10-17",
    "Statement": [
        {
            "Action": "*",
            "Resource": "*",
            "Effect": "Allow"
        }
    ]
}
```

This seems to allow CloudFormation to create/delete anything in an account including IAM resouces, even if I just wan't to deploy an API.

### Proposal

I hope that there were policy templates for tipical usecases such as an API or an S3 event handler.

**Comment by hoffa** [comment]
You can create an IAM role with reduced permissions and pass its ARN during the bootstrapping process.

**Comment by pjxiao** [comment]
@hoffa
Yes, the bootstrap subcommand allows us to pass a role ARN. But we need to create an IAM role in advance by ourselves. I feel that writing an IAM policy is one of the hardest parts of deployment. So I hope that the bootstrap subcommand allowed us to chose more restrictive version of an exceution role policy.

Thanks,

**Comment by Goblinlordx** [comment]
This seems a bit old. Can we at least consider denying IAM access to this role? It seems unsafe as it allows privilege escalation via creation of an administrative IAM role/user via cloudformation. I think if, at least, this is denied, it would provide a much more secure role. It might also be a good idea to maybe prevent allowing importing resources via cloudformation by default? Maybe this could be skipped but the IAM part seems almost necessary.
